challenge2/index.html

```diff
@@ -1,74 +1,223 @@
-                            <div class="input-wrapper">
-                                <img src="./assets/icons/user.svg" alt="user">
-                                <input type="text" id="username" name="username" placeholder="Enter your User name" required>
-                            </div>
-                        </div>
-
-                        <div class="input-group">
-                            <label for="password">Password</label>
-                            <div class="input-wrapper">
-                                <img src="./assets/icons/lock.svg" alt="lock">
-                                <input type="password" id="password" name="password" placeholder="Enter your password" required>
-                                <img src="./assets/icons/eye.svg" alt="eye" class="eyeImage">
-                            </div>
-                        </div>
-
-                        <div class="input-group">
-                            <label for="password">Confirm Password</label>
-                            <div class="input-wrapper">
-                                <img src="./assets/icons/lock.svg" alt="lock">
-                                <input type="password" id="passwordConfirm" name="passwordConfirm" placeholder="Confirm your password" required>
-                                <img src="./assets/icons/eye.svg" alt="eye" class="eyeImage">
-                            </div>
-                        </div>
-
-                        <button type="submit">Register</button>
-                    </form>
+              <div class="input-group">
+                <label for="passwordConfirm" id="confirmLabel"
+                  >Confirm Password</label
+                >
+                <div class="input-wrapper" id="confirmWrapper">
+                  <img src="./assets/icons/lock.svg" alt="lock" />
+                  <input
+                    type="password"
+                    id="passwordConfirm"
+                    name="passwordConfirm"
+                    placeholder="Confirm your password"
+                    required
+                  />
+                  <img
+                    src="./assets/icons/eye.svg"
+                    alt="eye"
+                    class="eyeImage"
+                  />
                 </div>
-            </div>
-        </section>
+                <span class="error-message" id="passwordConfirmError"></span>
+              </div>
+
+              <button type="submit">Register</button>
+            </form>
+          </div>
+        </div>
+      </section>
     </main>
 
-</body>
+    <script>
+      document.querySelectorAll(".eyeImage").forEach((eyeIcon) => {
+        eyeIcon.addEventListener("click", function () {
+          const passwordInput = this.previousElementSibling;
+          if (passwordInput.type === "password") {
+            passwordInput.type = "text";
+          } else {
+            passwordInput.type = "password";
+          }
```

Comment:
You can use the ternary operator here for cleaner syntax:

```suggestion
            passwordInput.type = passwordInput.type === "password" ? "text" : "password";
```